gt/main.c

```diff
@@ -1905,7 +1905,10 @@ init_gt_instances(struct gt_config *gt_conf)
 	 * in the fragmentation table via GGU packets.
 	 */
 	unsigned int num_mbuf = calculate_mempool_config_para("gt",
-		gt_conf->net, gt_conf->net->front.total_pkt_burst +
+		gt_conf->net, gt_conf->max_pkt_burst +
+		gt_conf->frag_max_entries +
+		(gt_conf->net->front.total_pkt_burst +
+		gt_conf->num_lcores - 1) / gt_conf->num_lcores +
 		gt_conf->max_pkt_burst + gt_conf->frag_max_entries);
```

Comment:
Isn't the last line a duplicate?

Reply:
No, they are for the GGU packets sent to Gatekeeper in the worst case.

Reply:
I didn't mean a duplicate in an implicit way, but literally:
```
..., gt_conf->max_pkt_burst +
gt_conf->frag_max_entries +
... +
gt_conf->max_pkt_burst + gt_conf->frag_max_entries);
```

Reply:
I see. These two are combined together.

Reply:
Ok, so it was not a duplicate. Could you extend the comment in the code to explain why we need to double the value `(gt_conf->max_pkt_burst + gt_conf->frag_max_entries)`? I don't get it. In my understanding, we need `gt_conf->max_pkt_burst` to read those packets from the queue of the NIC and `gt_conf->frag_max_entries` for the fragment packets.

Moreover, shouldn't we add `gt_conf->max_ggu_notify_pkts` to the sum as well?

Reply:
You are right. We should use ``gt_conf->max_ggu_notify_pkts`` instead of ``gt_conf->frag_max_entries`` for calculating the number of GGU packets. Even in the worst case, all the fragment packets (i.e., ``gt_conf->frag_max_entries`` packets) require to send GGU policies, we limit the maximum number of GGU packets by ``gt_conf->max_ggu_notify_pkts``.